locked-nft/contracts/LockedNFT.cdc

```diff
@@ -0,0 +1,154 @@
+import NonFungibleToken from "./NonFungibleToken.cdc"
+
+pub contract LockedNFT {
+
+    pub event ContractInitialized()
+    pub event NFTLocked(
+        id: UInt64,
+        to: Address?,
+        lockedAt: UInt64,
+        lockedUntil: UInt64,
+        duration: UInt64
+    )
+    pub event NFTUnlocked(
+        id: UInt64,
+        from: Address?
+    )
+
+    pub let CollectionStoragePath:  StoragePath
+    pub let CollectionPublicPath:   PublicPath
+
+    pub var totalLockedTokens:      UInt64
+    access(self) let lockedTokens:  {String: LockedData}
+
+    pub struct LockedData {
+        pub let id: UInt64
+        pub let owner: Address
+        pub let lockedAt: UInt64
+        pub let lockedUntil: UInt64
+        pub let duration: UInt64
```

Comment:
I don't know the UI or product requirements but it seems useless to store the `duration` if we are storing the `lockedUntil`. Only one would suffice

Reply:
I left this because it was an ask for NBA and extra data in an event is okay.